File: sorted_squared_array/sorted_squared_array.go
package main

import (
	"fmt"
	"sort"
)

func sortedSquaredArray(array []int) []int {
	squaredArray := make([]int, len(array))
	for i, value := range array {
		squaredArray[i] = value * value
	}
	sort.Ints(squaredArray)
	return squaredArray
}

func main() {
	result := sortedSquaredArray([]int{-2, -1, 1, 2, 3})
	fmt.Println(result) // [1 1 4 4 9]
}

File: sorted_squared_array/sorted_squared_array.ts
function sortedSquaredArray(array: number[]): number[] {
    const squaredArray = array.map(value => value * value);
    squaredArray.sort((a, b) => a - b);
    return squaredArray;
}

console.log(sortedSquaredArray([-2, -1, 1, 2, 3])); // [1, 1, 4, 4, 9]
